src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs

```diff
@@ -28,7 +28,7 @@ public class UserClaimsPrincipalFactory<TUser> : IUserClaimsPrincipalFactory<TUs
         ArgumentNullThrowHelper.ThrowIfNull(userManager);
         if (optionsAccessor == null || optionsAccessor.Value == null)
         {
-            throw new ArgumentNullException(nameof(optionsAccessor));
+            throw new ArgumentException($"{nameof(optionsAccessor)} cannot be null or wrap a null value.", nameof(optionsAccessor));
```

Comment:
Seems like this should actually be `ArgumentNullException` since `optionsAccessor` is known to be null here...

Reply:
Yes, I guess the most correct would be to split into two exceptions. First when optionsAccessor is null, and an ArgumentException when "only" the 'optionAccessor.Value' is null (since the argument is not actually null).